### ISSUE TITLE
Literal type not working in VS Code

### ISSUE THREAD

**Issue by lukjoh** [microsoft/pyright#4184]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

In VS Code, with the latest Pylance extension installed, assigning a Literal type to a variable results in "Unknown" rather than the Literal type.

For example:
    X = Literal[1, 2, 3]
    x: X = 1

results in the following error:
    "Type of "x" is unknown  Pylance(ReportUnknownVariableType)

**To Reproduce**
Steps to reproduce the behavior.

Pylance installed in VS Code

**Expected behavior**

The above code example should work according to the documentation at:
https://docs.python.org/3/library/typing.html#typing.Literal

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

![Error](https://user-images.githubusercontent.com/59101551/201475157-904e502e-75ad-4389-a13f-3cac1d627b09.png)

```python

X = Literal[1, 2, 3]
x: X = 1
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by erictraut** [comment]
The `reportUnknownVariableType` is a strict-mode check that is not on by default. I presume that you have explicitly enabled it (or implicitly enabled it by setting typeCheckingMode to "strict"). 

Can you provide a complete, self-contained sample? `Literal` is not a builtin symbol. How are you importing it?

I presume you're importing it from `typing` like this?

```python
from typing import Literal

X = Literal[1, 2, 3]
x: X = 1
```

This works fine for me without producing any errors. Can you confirm that you receive errors if you create a new source file and paste the above code sample into it.

**Comment by lukjoh** [comment]
Hi Eric,
Thank you very much for your time and prompt response. 
I found the issue and it was my own silly fault. For some reason I had the type definition sitting inside a function (I was very tired and must not have been thinking straight). 
This issue can be closed.
Sorry for wasting your time.

**Comment by erictraut** [comment]
It looks like you're attempting to create a type alias called `X`. In Python, the expression form for a type alias overlaps that of a regular variable assignment, so type checkers need to use a variety of heuristics to tell the two apart. The maintainers of various Python type checkers (mypy, pyre, pyright, etc.) have discussed this issue and attempted to adopt similar heuristics. One such rule is that type aliases cannot be declared in function scopes. They are allowed only in global or class scopes. So in your example, `X = Literal[1, 2]` is not interpreted as a type alias declaration but a regular variable assignment. That means `X` cannot be used as a type annotation because it's a variable.

FWIW, I've authored [PEP 695](https://peps.python.org/pep-0695/) that proposes to add an official keyword to Python 3.12 for declaring type aliases. If this proposal is accepted, it would allow you to declare type aliases within a function scope. Here's how that would look.

```python
from typing import Literal

def run():
    type X = Literal[1, 2]
    x: X = 1
```

**Comment by lukjoh** [comment]
That makes sense. 
Thank you.
